src/battleship/Battleship.java

```diff
@@ -63,88 +79,115 @@ public int colNum(String coordinate) {
+//
+//            if (coordinate.isEmpty()) {
+//                System.out.println("Invalid input. Try again.");
+//                continue;
+//            }
+//
+//            int row = rowNum(coordinate.get());
+//            int col = colNum(coordinate.get());
+//
+//            if (row < 0 || col < 0) {
+//                System.out.println("Invalid location. Try again.");
+//                continue;
+//            }
+//
+//            try {
+//                if (board.isShip(row, col)) {
+//                    hiddenBoard.setIndex(row, col, 'X');
+//                    board.setIndex(row, col, 'X');
+//                    hiddenBoard.printField();
+//                    for (Ship ship : ships) {
+//                        board.isSunken(ship);
+//                    }
+//                    boolean sank = false;
+//                    for (Ship type : ships) {
+//                        if (type.isSunken()) {
+//                            sank = true;
+//                            ships.remove(type);
+//                            break;
+//                        }
+//                    }
+//                    if (sank) {
+//                        System.out.println("You sank a ship! Specify a new target:");
+//                    } else {
+//                        System.out.println("You hit a ship! Try again:");
+//                    }
+//                    oCounter--;
+//                } else if (board.isEmpty(row, col)) {
+//                    hiddenBoard.setIndex(row, col, 'M');
+//                    board.setIndex(row, col, 'M');
+//                    hiddenBoard.printField();
+//                    System.out.println("You missed. Try again:");
+//                } else if (board.isHit(row, col)) {
+//                    hiddenBoard.printField();
+//                    System.out.println("You already hit this one. Try again:");
+//                } else if (board.isMiss(row, col)) {
+//                    hiddenBoard.printField();
+//                    System.out.println("You already missed this one. Try again:");
+//                }
+//            } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException e) {
+//                System.out.println("Error! You entered the wrong coordinates! Try again:");
+//            }
+//        }
+//        System.out.println("You sank the last ship. You won. Congratulations!");
+    }
+
+    public void setPlayerBoard(List<Ship> player, Board board) {
+        player = new ArrayList<>(Arrays.asList(
+                new Ship(5, "Aircraft Carrier"),
+                new Ship(4, "Battleship"),
+                new Ship(3, "Submarine"),
+                new Ship(3, "Cruiser"),
+                new Ship(2, "Destroyer")
         ));
 
-        for (final Ship ship : ships) {
+        for (final Ship ship : player) {
             board.printField();
             setShipType(ship, board);
         }
+    }
```

Comment:
This is why we make parameters final. Assigning to this parameter has no useful effect. The parameter goes out of scope at the end of this method and the list goes away.

It looks like you really don't need the parameter at all and you really meant to say.

```java
    public void setPlayerBoard(final List<Ship> playerShips, final Board board) {
        final List<Ship> ships = Arrays.asList(
            new Ship(5, "Aircraft Carrier"),
            new Ship(4, "Battleship"),
            new Ship(3, "Submarine"),
            new Ship(3, "Cruiser"),
            new Ship(2, "Destroyer")
        );

        playerShips.addAll(ships));

        for (final Ship ship : ships) {
            board.printField();
            setShipType(ship, board);
        }
    }
```

Reply:
Hey Talden, how often should I be making parameters and variable final? In this case, I'm assuming that it's ok to make the List and Board objects final because I wouldn't be making any direct modifications to those.